src/deadline_worker_agent/startup/bootstrap.py

```diff
@@ -573,13 +633,14 @@ def _get_metadata(metadata_type: str) -> requests.Response | None:
         response = requests.put(
             "http://169.254.169.254/latest/api/token",
             headers={"X-aws-ec2-metadata-token-ttl-seconds": "30"},
+            timeout=5,  # Non-aws worker hosts will time-out, but the default timeout can be > 20s
```

Comment:
I believe this could be shortened. IMDS should be very quick to establish a connection if available and this would slow down the worker agent startup for on-premise or off-AWS workers.

How about half a second?

```suggestion
            timeout=0.5,  # Non-aws worker hosts will time-out, but the default timeout can be > 20s
```

Reply:
Yeah, I think half a second is still generous. Updated to this in latest commit.